### PR TITLE
Add the ability to provide identity as a UUID

### PR DESCRIPTION
**Pull request by t-persson**
### Applicable Issues

fixes: eiffel-community/etos#73

### Description of the Change

Add the ability to start ETOS using artifact ID instead of PackageURL string for more explicit IUT selection.

### Alternate Designs

I opted to keep the packageurl imeplementation instead of removing as it would be a huge breaking change.

### Benefits

We can start tests on a more explicit IUT, instead of relying on the package URL being unique.

### Possible Drawbacks

None that I can think of.

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by: Tobias Persson <tobias.persson@axis.com>

### CODE REVIEW

File: src/etos_client/__main__.py [hunk 1]
```diff
@@ -70,14 +68,14 @@ def parse_args(args):
     parser.add_argument(
         "-i",
         "--identity",
-        help="Artifact created identity purl to execute test suite on.",
-        **environ_or_required("IDENTITY")
+        help="Artifact created identity purl or ID to execute test suite on.",
+        **environ_or_required("IDENTITY"),
```
Comment (fredjn) [inline src/etos_client/__main__.py:72]: This should be IDENTIFIER. Or add check for environment variable IDENTIFIER to not break backwards compatibility.

File: src/etos_client/__main__.py [hunk 2]
```diff
@@ -284,11 +282,10 @@ def main(args):
 
     setup_logging(args.loglevel)
     info = generate_spinner(args.no_tty)
-    identity = PackageURL.from_string(args.identity)
 
     for key, value in args._get_kwargs():  # pylint:disable=protected-access
         etos.config.set(key, value)
-    etos.config.set("identity", identity)
+    etos.config.set("identity", args.identity)
```
Comment (fredjn) [inline src/etos_client/__main__.py:288]: This should be called identifier or artifact_identifier.

File: src/etos_client/client.py [hunk 3]
```diff
@@ -49,6 +83,16 @@ def data(self):
             "log_area_provider": self.etos.config.get("log_area_provider"),
             "test_suite_url": self.etos.config.get("test_suite"),
         }
+        identity = self.etos.config.get("identity")
```
Comment (fredjn) [inline src/etos_client/client.py:86]: artifiact_identifier

**Review by fredjn — CHANGES_REQUESTED** [review]
All comment relate to my suggestion to change identity -> artifact_identifier. All changes relating to this should be done without breaking backwards compatibility (if possible).

File: src/etos_client/__main__.py (hunk 1 above)
Comment (t-persson) [reply src/etos_client/__main__.py:72]: I will not do this rename. Let's write an issue to fix this in the future. This is a breaking change and requires deprecation.

Comment (fredjn) [reply src/etos_client/__main__.py:72]: Agreed!

**Review by fredjn — APPROVED** [review]

**Review by fredjn — APPROVED** [review]

**Review by andjoe-axis — APPROVED** [review]

**Pull request merged**